Add an aria-label to the editor blocks.

afercia opened the pull request:
First pass to try to add an aria-label to the Editor blocks. Code and wording could be improved, any feedback welcome.

All the editor blocks need to be labeled to allow screen reader users to get `what` a block is.

Screenshots:

![screen shot 2017-06-30 at 17 55 40](https://user-images.githubusercontent.com/1682452/27744056-6da56ed8-5dbe-11e7-88c2-5096ddf04337.png)

![screen shot 2017-06-30 at 17 57 09](https://user-images.githubusercontent.com/1682452/27744057-6dababae-5dbe-11e7-8ff2-9e98cb6d64af.png)

Re: VoiceOver in the screenshot above:
- "group" is added by VoiceOver because blocks have no semantic role, so it announces them as generic "group"
- also the number of items in the group is added by VoiceOver


Fixes #562

afercia committed:
Add an aria-label to the editor blocks.

youknowriad commented on editor/modes/visual-editor/block.js:
```diff
@@ -236,6 +237,7 @@ class VisualEditorBlock extends Component {
 	render() {
 		const { block, multiSelectedBlockUids } = this.props;
 		const blockType = getBlockType( block.name );
+		const blockLabel = sprintf( __( 'Editor block: %s' ), blockType.title );
```
Maybe I'd just use `Block: %s` but I'm fine either way

youknowriad reviewed (approved):
LGTM 👍

afercia committed:
Simplify label and add translators comment.

afercia commented:
I've simplified the aria label to be just `Block: %s`. Also, added a translators comment, using the same one used for the block mover label. That could be probably improved a bit. Worth noting these translators comments should preferably be inside `sprintf()`. I see this has already been considered and there are thoughts to improve a bit the tooling, see https://github.com/WordPress/gutenberg/pull/984#discussion_r120464365
Not sure if there's already a specific GH issue.

aduth commented:
>Edit: seems it has already been addressed? /cc @aduth

Yes, should be improved as of #1068
